sdk/src/account_api.rs

```diff
@@ -0,0 +1,295 @@
+//! Defines account apis for use in Solana programs.
+
+use crate::credit_debit_account::KeyedCreditDebitAccount;
+use crate::credit_only_account::KeyedCreditOnlyAccount;
+use crate::instruction::InstructionError;
+use crate::pubkey::Pubkey;
+use std::fmt::Debug;
+use std::sync::atomic::Ordering;
+
+#[derive(Debug)]
+pub enum AccountWrapper<'a> {
+    CreditDebit(KeyedCreditDebitAccount<'a>),
+    CreditOnly(KeyedCreditOnlyAccount<'a>),
+}
+
+pub trait AccountApi: Debug {
+    fn signer_key(&self) -> Option<&Pubkey>;
+    fn unsigned_key(&self) -> &Pubkey;
+    fn lamports(&self) -> u64;
+    fn set_lamports(&mut self, lamports: u64) -> Result<(), InstructionError>;
+    fn credit(&mut self, lamports: u64) -> Result<(), InstructionError>;
+    fn debit(&mut self, lamports: u64) -> Result<(), InstructionError>;
+    fn get_data(&self) -> &[u8];
```

Comment:
Can this be `data()` instead of `get_data()`?

Reply:
Sure enough